packages/compiler/core/helpers/type-utils.ts

```diff
@@ -0,0 +1,135 @@
+import {
+  Enum,
+  Interface,
+  Model,
+  ModelProperty,
+  ModelStatementNode,
+  Namespace,
+  Operation,
+  Type,
+} from "../types.js";
+
+export interface TypeNameOptions {
+  namespaceFilter: (ns: Namespace) => boolean;
+}
+
+export function getTypeName(type: Type, options?: TypeNameOptions): string {
+  switch (type.kind) {
+    case "Namespace":
+      return getNamespaceString(type, options);
+    case "TemplateParameter":
+      return type.node.id.sv;
+    case "Model":
+      return getModelName(type, options);
+    case "ModelProperty":
+      return getModelPropertyName(type, options);
+    case "Interface":
+      return getInterfaceName(type, options);
+    case "Operation":
+      return getOperationName(type, options);
+    case "Enum":
+      return getEnumName(type, options);
+    case "EnumMember":
+      return `${getEnumName(type.enum, options)}.${type.name}`;
+    case "Union":
+      return type.name || type.options.map((x) => getTypeName(x, options)).join(" | ");
+    case "UnionVariant":
+      return getTypeName(type.type, options);
+    case "Tuple":
+      return "[" + type.values.map((x) => getTypeName(x, options)).join(", ") + "]";
+    case "String":
+    case "Number":
+    case "Boolean":
+      return type.value.toString();
+    case "Intrinsic":
+      return type.name;
+  }
+
+  return "(unnamed type)";
+}
+
+/**
+ * Return the full name of the namespace(e.g. "Foo.Bar")
+ * @param type namespace type
+ * @param options
+ * @returns
+ */
+export function getNamespaceString(type: Namespace, options?: TypeNameOptions): string {
```

Comment:
This name is pretty non-descriptive, unsure if we want to clean it up as part of this PR? (Could rename and deprecate the old).

Reply:
open to that, what were you thinking as a new name 
- `getNamespaceName`
- `getNamespaceFullName`

if we are deprecating `getNamespaceString` in the `checker` is it worth adding the new one as well in the checker or user should just use that helper. And then should we deprecate `getTypeName` in the checker too.

Reply:
Ok renamed to `getNamespaceFullName`